### ISSUE TITLE
Markdown Support

### ISSUE THREAD

**Issue by aaronweaver** [DefectDojo/django-DefectDojo#173]
Add Markdown support and an editor to product description, vulnerability and other text fields in Dojo.

**Comment by grendel513** [comment]
@aaronweaver @devGregA - this one is huge, and will affect report generation as the markdown will have to be interpreted and rendered before the PDFs and/or ASCIIDoc are generated.  Also need to consider future report formats like CSV, XML, JSON, etc since some characters may not be valid for the formats.

**Comment by devGregA** [comment]
Do we want to make this a swag issue? @grendel513 @aaronweaver ? XD

**Comment by humor4fun** [comment]
Perhaps a stopgap solution would be to build the UI for it to be written/displayed and then when report gen happens, just strip the markdown out and replace them with appropriate comma, space or newline characters.

**Comment by aaronweaver** [comment]
Couldn't we use one of the Python Markdown render to do that? For the report's the PDF is essentially a screen scrape of the UI so rendering in HTML would be ok. For ASCII there's probably a markdown strip or converter of some sort.

**Comment by patriknordlen** [comment]
I've worked on this and have a version with Markdown for the fields. In order to provide compatibility I've added a "formatting" field in the Finding model which can be either "raw" or "markdown", with "raw" being the default value so legacy findings and reports are not affected.

I looked at various Markdown modules for Django and ended up with MarkdownX (https://pypi.python.org/pypi/django-markdownx), mostly because the inbuilt preview functionality and ability to add images by dragging and dropping. I'm missing stuff that other modules have, like an editor with a menu for the most common stuff like adding bold text, italic text, bullet lists etc, but no package has both that functionality and the functionality found in MarkdownX.
